src/modules/configuration/src/lib/Configuration.c

```diff
@@ -0,0 +1,545 @@
+            }
+            
+            if (JSONSuccess == json_object_set_number(jsonObject, g_fullLoggingEnabledName, (double)(fullLoggingEnabled ? 1: 0)))
+            {
+                g_fullLoggingEnabled = fullLoggingEnabled;
+            }
+            else
+            {
+                OsConfigLogError(ConfigurationGetLog(), "json_object_set_boolean(%s, %s) failed", g_fullLoggingEnabledObject, fullLoggingEnabled ? "true" : "false");
+            }
+
+            if (JSONSuccess == json_object_set_number(jsonObject, g_commandLoggingEnabledName, (double)(commandLoggingEnabled ? 1 : 0)))
+            {
+                g_commandLoggingEnabled = commandLoggingEnabled;
+            }
+            else
+            {
+                OsConfigLogError(ConfigurationGetLog(), "json_object_set_boolean(%s, %s) failed", g_commandLoggingEnabledObject, commandLoggingEnabled ? "true" : "false");
+            }
+            
+            if (JSONSuccess == json_object_set_number(jsonObject, g_iotHubProtocolName, (double)iotHubProtocol))
+            {
+                g_iotHubProtocol = iotHubProtocol;
+            }
+            else
+            {
+                OsConfigLogError(ConfigurationGetLog(), "json_object_set_number(%s, %d) failed", g_iotHubProtocolObject, iotHubProtocol);
+            }
+        }
+
+        if (MMI_OK == status)
+        {
+            newConfiguration = json_serialize_to_string_pretty(jsonValue);
+
+            if (newConfiguration)
+            {
+                if (SavePayloadToFile(g_configurationFile, newConfiguration, strlen(newConfiguration), ConfigurationGetLog()))
+                {
+                    if (false == RestartDaemon(g_osConfigDaemon, ConfigurationGetLog()))
+                    {
+                        OsConfigLogError(ConfigurationGetLog(), "Failed restarting %s to apply configuration", g_osConfigDaemon);
+                        status = ESRCH;
+                    }
+                }
+                else
+                {
+                    OsConfigLogError(ConfigurationGetLog(), "Failed saving configuration to %s", g_osConfigConfigurationFile);
+                    status = ENOENT;
+                }
+            }
+            else
+            {
+                OsConfigLogError(ConfigurationGetLog(), "json_serialize_to_string_pretty failed");
+                status = EIO;
+            }
+        }
+    }
+
+    if (MMI_OK == status)
+    {
+        OsConfigLogInfo(ConfigurationGetLog(), "New configuration successfully applied: %s", IsFullLoggingEnabled() ? newConfiguration : "-");
+    }
+    else
+    {
+        OsConfigLogError(ConfigurationGetLog(), "Failed to apply new configuration: %s", IsFullLoggingEnabled() ? newConfiguration : "-");
+    }
+        
+    if (jsonValue)
+    {
+        json_value_free(jsonValue);
+    }
+
+    FREE_MEMORY(existingConfiguration);
+    FREE_MEMORY(newConfiguration);

```

Comment:
>FREE_MEMORY(newConfiguration); [](http://example.com/codeflow?start=4&length=30)

Is it preferred to free this using `json_free_serialized_string` instead? As it was created with `json_serialize_to_string_pretty`.

Reply:
Makes sense, thanks!